UPGRADE.md

```diff
@@ -1,5 +1,9 @@
 # Upgrade to 2.11
 
+## Deprecated the usage of Abstraction\Result as Driver\Result
+
+The usage of the `Abstraction\Result` as a sub-type of the `Driver\Result` interface is deprecated.
```

Comment:
```suggestion
The usage of the `Abstraction\Result` as a sub-type of the `Driver\Result` interface is deprecated. The interface `Driver\Result` will be removed from `Abstraction\Result` in 3.0.
```

Reply:
The "will be removed" removed wording is a bit confusing: will we also remove the methods or not? Would it make sense to reword it as "the interface X will not extend Y in 3.0"?

Reply:
Sounds better than my text, yes. :+1: